### DISCUSSION TITLE
Rendering not the same on different PyVista versions

### DISCUSSION THREAD

**Discussion by juaniio** [pyvista/pyvista#2929] [Q&A]
Hi! I'm running the exact same code on two different computers:

**Computer 1**: `Ubuntu 16.04`, `Python 3.5`, `PyVista 0.26.1`.
**Computer 2**: `Ubuntu 22.04`, `Python 3.10`, `PyVista 0.34.2`.

The camera `position` and `viewup` vectors are set equally:
` {'position': [1, 1, 1], 'viewup': [0, 1, 0]}`

And as I said, the code is the same. It just loads `mesh_vertices`, `mesh_faces`, `texture_array`, and `texture_coords` from the same `.npy` files, and then runs:

```
# Convert from texture array to PyVista texture:
texture = pv.numpy_to_texture(texture_array)
texture.SetInterpolate(True)

mesh = pv.PolyData(mesh_vertices, mesh_faces)
# Add texture coords to PyVista mesh:
mesh.t_coords = texture_coords

pv_plotter = pv.Plotter()
pv_plotter.set_background('white')

# Add mesh with texture to plotter:
pv_plotter.add_mesh(mesh, texture=texture, lighting=False)

image = pv_plotter.screenshot()
```

But the results are different. The rendering is not the same (the rendering on **Computer 2** is **not** what I want, it seems to be a little skewed).

**Computer 1 (what I want)**:
![correct](https://user-images.githubusercontent.com/7669485/176959504-75cf0b63-e512-41f3-9e31-a3504faf3ace.png)

**Computer 2 (skewed)**:
![skewed](https://user-images.githubusercontent.com/7669485/176959510-64b95557-e522-4e9a-a0fe-8bf8023e9987.png)

Any thoughts? Thanks in advance.

**Comment by adeak** [comment]
> The camera position and viewup vectors are set equally:
> `{'position': [1, 1, 1], 'viewup': [0, 1, 0]}`

Could you please
1. add these explicitly to your plotting script to ensure they are the same, and
2. check that both versions also have the same focal point for the camera?

**Reply by juaniio** [reply]
I just checked `position` and `viewup` in both versions, and they are indeed different. 

What's happening apparently is that the `pv_plotter.add_mesh` function resets the position of the plotter's camera (I didn't set the plotter's camera position explicitly before).

My understanding is that in the different versions this automatic setup of camera position is behaving differently.

With the same `mesh`, `texture_array`, and `texture_coords`, this code prints a different plotter's camera `position` for the two versions:
```
import pyvista as pv
pv.rcParams['camera']['viewup'] = [0, 1, 0]

# Convert from texture array to PyVista texture:
texture = pv.numpy_to_texture(texture_array)
texture.SetInterpolate(True)

mesh = pv.PolyData(mesh_vertices, mesh_faces)
# Add texture coords to PyVista mesh:
mesh.t_coords = texture_coords

pv_plotter = pv.Plotter()
pv_plotter.set_background('white')

# Add mesh with texture to plotter:
pv_plotter.add_mesh(mesh, texture=texture, lighting=False)

print(pv_plotter.camera.GetPosition())
print(pv_plotter.camera.GetViewUp())
print(pv_plotter.camera.GetFocalPoint())
```

**Version 1 output** (with `PyVista 0.26.1`):
```
(29.0, -48.0, 216.72772492742433)
(0.0, 1.0, 0.0)
(29.0, -48.0, 0.05)
```

**Version 2 output** (with `PyVista 0.34.2`):
```
(154.09894281424414, 77.09894281424414, 125.14894281424414)
(0.0, 1.0, 0.0)
(29.0, -48.0, 0.05)
```

I do want this automatic camera setup, in `PyVista 0.26.1` it always worked as I expected, setting the camera in a way that fits the different objects that I'm going to render with that plotter, one by one. I'm just trying to migrate from `0.26.1` to `0.34.2`.

Additionally, only in `PyVista 0.34.2`, if I just call `pv_plotter.camera.GetPosition()` before adding the mesh to the plotter, the automatic setup of the camera does not take place. In `PyVista 0.26.1` the position is still automatically set to the same position included above: `(29.0, -48.0, 216.72772492742433)`.

**Reply by adeak** [reply]
I'm not familiar with these mechanics in PyVista, so a quick guess: please check if passing `reset_camera=False` to `plotter.add_mesh()` does what you expect.

(And you're probably aware already, but just to be sure: we now have a dedicated `Camera` class with nice pythonic properties such as `plotter.camera.position`, see https://docs.pyvista.org/api/core/_autosummary/pyvista.Camera.html?highlight=camera#pyvista.Camera)

**Reply by akaszynski** [reply]
I actually addressed this (partially) in #2925. It looks like we almost always reset the camera when adding meshes.
